dataset/src/main/scala/frameless/TypedWindow.scala

```diff
@@ -0,0 +1,156 @@
+package frameless
+
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.catalyst.expressions.{ UnspecifiedFrame, WindowFrame }
+import org.apache.spark.sql.expressions.{ Window, WindowSpec }
+import shapeless.ops.hlist.{ Mapper, ToTraversable }
+import shapeless.{ HList, ProductArgs }
+
+trait OrderedWindow
+trait PartitionedWindow
```

Comment:
Have you considered using a HKT instead?

Reply:
I'm not sure what you mean by that here. I use these traits to ensure that a partition or an order has been assigned to a window as some window functions require one or both to be present (eg `denseRank` needs an ordering). An ordering might be on multiple disparate types and has no affect on the return type of the window. What would be gained by making `OrderedWindow` a HKT here or do you mean something else?